src/openvic-simulation/map/Province.cpp

```diff
@@ -58,9 +58,10 @@ bool Province::load_positions(BuildingTypeManager const& building_type_manager,
 	return ret;
 }
 
-bool Province::expand_building(std::string_view building_type_identifier) {
-	BuildingInstance* building = buildings.get_item_by_identifier(building_type_identifier);
+bool Province::expand_building(size_t building_index) {
```

Comment:
in the same way you made color strongly typed, you might consider making this integer strongly typed

Reply:
I'm not planning on using integer indices to identify province buildings internally, just for communication with the UI which needs to use primitive types for the actual godot-cpp interface functions, so a stong type wouldn't have much use. `Province::index_t` would be a good candidate for making a strong type, especially since there are a couple other `index_t` type aliases with different sizes.